gef.py

```diff
@@ -3771,7 +3771,11 @@ def reset_architecture(arch: Optional[str] = None, default: Optional[str] = None
             raise OSError(f"Specified arch {arch.upper()} is not supported")
 
     if not gef.binary:
-        gef.binary = get_elf_headers()
+        try:
+            gef.binary = get_elf_headers()
+        except RuntimeError:
+            # in case the binary is not an ELF file
```

Comment:
GEF is meant for Linux & ELF. Other OS/formats are not supported in vanilla GEF, you can use `gef-extras` for that purpose.

Reply:
Ok, I will take a look at `gef-extras`.